packages/commonwealth/server/util/requirementsModule/requirementsSchema.json

```diff
@@ -0,0 +1,95 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "definitions": {
+    "ContractSource": {
+      "type": "object",
+      "properties": {
+        "source_type": {
+          "type": "string",
+          "enum": ["erc20", "erc721"]
+        },
+        "chain_id": {
+          "type": "string"
+        },
+        "contract_address": {
+          "type": "string"
+        }
+      },
+      "required": ["source_type", "chain_id", "contract_address"]
+    },
+    "NativeSource": {
+      "type": "object",
+      "properties": {
+        "source_type": {
+          "type": "string",
+          "enum": ["eth_native"]
+        },
+        "chain_id": {
+          "type": "string"
+        }
+      },
+      "required": ["source_type", "chain_id"]
+    },
+    "CosmosSource": {
+      "type": "object",
+      "properties": {
+        "source_type": {
+          "type": "string",
+          "enum": ["cosmos_native"]
+        },
+        "chain_id": {
+          "type": "string"
+        },
+        "token_symbol": {
+          "type": "string"
+        }
+      },
+      "required": ["source_type", "chain_id", "token_symbol"]
+    },
+    "ThresholdData": {
+      "type": "object",
+      "properties": {
+        "threshold": {
+          "type": "string"
+        },
+        "source": {
+          "oneOf": [
+            { "$ref": "#/definitions/ContractSource" },
+            { "$ref": "#/definitions/NativeSource" },
+            { "$ref": "#/definitions/CosmosSource" }
+          ]
+        }
+      },
+      "required": ["threshold", "source"]
+    },
+    "AllowlistData": {
```

Comment:
We will want additional validation on allow lists, i.e. that provided addresses are formatted in a valid way (and we will want that for contract addresses too). Maybe we can use a regex, like I did in the schema I provided on the planning doc?

Reply:
Are cosmos addresses formatted the same as EVM addresses?

Reply:
They're not -- they're bech32 and will require a different regex.